docs/source/examples/e2e_flow.rst

```diff
@@ -0,0 +1,438 @@
+
+
+Once the config is updated, let's kick off evaluation! We'll use the
+``truthfulqa_mc2`` task which is also the default in the config.
+
+.. code-block:: bash
+
+    tune run eleuther_eval --config ./custom_eval_config.yaml
+
+
+Once evaluation is complete, you'll see the following in the logs.
+
+.. code-block:: bash
+
+    [evaluator.py:324] Running loglikelihood requests
+    [eleuther_eval.py:195] Eval completed in 121.27 seconds.
+    [eleuther_eval.py:197] truthfulqa_mc2: {'acc,none': 0.48919958543950917 ...}
+
+So seems like our fine-tuned model gets ~48% on this task. Which is pretty good.
+An exercise for you to do is to modify the config and run this eval using the
+original model from HF. You should get somewhere around 39-40%.
+
+|
+
+Generation!
+-----------
+
+We've run some evaluations and the model seems to be doing well. But does is really
+generate meaningful text for the prompts you care about? Let's find out!
+
+For this, we'll use the
+`generate recipe <https://github.com/pytorch/torchtune/blob/main/recipes/generate.py>`_
+and the associated
+`config <https://github.com/pytorch/torchtune/blob/main/recipes/configs/generate.yaml>`_.
+
+
+Let's first copy over the config to our local working director so we can make changes.
+
+.. code-block:: bash
+
+    tune cp generate ./custom_generation_config.yaml
+
+Let's modify ``custom_generation_config.yaml`` to include the following changes.
+
+.. code-block:: yaml
+
+    checkpointer:
+        _component_: torchtune.utils.FullModelHFCheckpointer
+
+        # directory with the checkpoint files
+        # this should match the output_dir specified during
+        # finetuning
+        checkpoint_dir: <checkpoint_dir>
+
+        # checkpoint files for the fine-tuned model. This should
+        # match what's shown in the logs above
+        checkpoint_files: [
+            hf_model_0001_0.pt,
+            hf_model_0002_0.pt,
+        ]
+
+        output_dir: <checkpoint_dir>
+        model_type: LLAMA2
+
+    # Make sure to update the tokenizer path to the right
+    # checkpoint directory as well
+    tokenizer:
+        _component_: torchtune.models.llama2.llama2_tokenizer
+        path: <checkpoint_dir>/tokenizer.model
+
+
+Once the config is updated, let's kick off generation! We'll use the
+default settings for sampling with ``top_k`` set to ``300`` and a
+temperature of ``0.8``. These are standard settings for Llama2 7B and
+we recommend inspecting the model with these before playing around with
+these parameters.
+
+We'll use a different prompt from the one in the config
+
+.. code-block:: bash
+
+    tune run generate \
+    --config generate \
+    prompt="What are some interesting sites to visit in th Bay Area?"
+
+
+Once generation is complete, you'll see the following in the logs.
+
+
+.. code-block:: bash
+
+    [generate.py:92] Exploratorium in San Francisco has made the cover of Time Magazine,
+                     and its awesome. And the bridge is pretty cool...
```

Comment:
is this the actual model output? because it is grammatically incorrect :)
```suggestion
                     and it's awesome. And the bridge is pretty cool...
```